server/message_dispatch.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2015 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"strings"
+	"sync"
+)
+
+type MessageDispatch struct {
+	input       chan string
+	output      map[string][]chan string
```

Comment:
Perhaps I don't get it, but first subscriber can get Mutex and Insert Cond in map. When event arrives, dispatcher Broadcasts and removes Cond from map. GC removes it after all client goroutines remove it from scope.